src/garage/sampler/is_sampler.py

```diff
@@ -54,31 +56,58 @@ def __init__(self,
         self._hist = []
         self._is_itr = init_is
 
-        super().__init__(algo, env)
+        self.algo = algo
+        self.env = env
 
     @property
     def history(self):
-        """
+        """History of policies.
+
         History of policies that have interacted with the environment and the
-        data from interaction episode(s)
+        data from interaction episode(s).
+
+        Returns:
+            list: a list of paths.
+
         """
         return self._hist
 
     def add_history(self, policy_distribution, paths):
-        """
-        Store policy distribution and paths in history
+        """Store policy distribution and paths in history.
+
+        Args:
+            policy_distribution (object): Policy distribution.
+            paths (list): Paths.
+
         """
         self._hist.append((policy_distribution, paths))
 
     def get_history_list(self, n_past='all'):
-        """
-        Get list of (distribution, data) tuples from history
+        """Get list of (distribution, data) tuples from history.
+
+        Args:
+            n_past (str): Number of past policies to update from
```

Comment:
does this also accept ints? if so it needs to be documented. Perhaps you can change the function to accept only ints and use `None` for all? That will make the documentation easier to write and the interface clearer.